src/Channels.Networking.Libuv/UvTcpTransportProvider.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Channels.Networking.Libuv
+{
+    /// <summary>
+    /// A TransportProvider using libuv
+    /// </summary>
+    public class UvTcpTransportProvider : TransportProvider
+    {
+        UvThread thread = new UvThread();
```

Comment:
An interesting thing about this one is that for the server, you probably want to have multiple threads (or at least a setting) so putting client and server on the same interface has some interesting side effects like this. Clients also probably need things like connection pooling and caching based on DNS name or whatever groups it deeps appropriate (ServicePointManager anyone?)